src/models/geograph.py

```diff
@@ -168,29 +164,26 @@ def __init__(
         # Save resulting graph, if we didn't load from graph.
         if not load_from_graph and graph_save_path is not None:
             graph_save_path = pathlib.Path(graph_save_path)
-            if graph_save_path.suffix not in (".pickle", ".pkl", ".gz", ".bz2"):
-                raise ValueError(
-                    """Argument `graph_save_path` should be a pickle file or
-                    compressed file."""
-                )
             os.makedirs(graph_save_path.parent, exist_ok=True)
-            self._save_graph(graph_save_path)
+            self.save_graph(graph_save_path)
+
+        self.components = self.get_graph_components(calc_polygons=False)
 
         print(
             f"Graph successfully loaded with {self.graph.number_of_nodes()} nodes",
             f"and {self.graph.number_of_edges()} edges.",
         )
 
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, GeoGraph):
+            return False
+        return nx.is_isomorphic(self.graph, o.graph)
```

Comment:
Yeah I just realised it's exponential in the number of vertices - for some reason that function is still very fast when comparing graphs which have quite a few differences though, but impossibly slow for graphs which are the same. I've changed it to `nx.fast_could_be_isomorphic` which takes about half a second to compare even for graphs which are equal. It's only an approximation of course but I think it will be quite accurate for the differences usually found between GeoGraphs.